pytket/extensions/cutensornet/mps/mps.py

```diff
@@ -45,6 +45,59 @@ class DirectionMPS(Enum):
     RIGHT = 1
 
 
+class CuTensorNetHandle:
+    """Initialise the cuTensorNet library with automatic workspace memory
+    management.
+
+    Note:
+        Always use as ``with CuTensorNetHandle() as libhandle:`` so that cuTensorNet
+        handles are automatically destroyed at the end of execution.
+
+    Attributes:
+        handle (int): The cuTensorNet library handle created by this initialisation.
+        device_id (int): The ID of the device (GPU) where cuTensorNet is initialised.
+            If not provided, defaults to ``cp.cuda.Device()``.
+    """

```

Comment:
Should also add "Raises: " field.

Reply:
My understanding is that these will almost never be raised: it's unlikely that the user will have cuQuantum running on an old enough GPU with an old enough driver. And even if they do, I believe that the error message they would get is already informative enough. Therefore, I don't think there's a point to document it and it's actually detrimental, adding superflous information for most users.

Reply:
Agreed!